src/elf/resolver.rs

```diff
@@ -111,8 +127,22 @@ impl FileCache<ElfResolverData> {
         // SANITY: We made sure to create the desired resolver above.
         Ok(resolver.unwrap())
     }
-}
 
+    /// Register an existing `ElfResolver` for a given path.
+    pub(crate) fn register(&self, path: &Path, elf_resolver: Rc<ElfResolver>) -> Result<()> {
+        // TODO: Need to use passed-in `ElfResolver`'s file instead of
+        //       opening a new one here. Doing so will need adjustments
+        //       to `FileCache` API.
+        let (_file, cell) = self.entry(path.actual_path())?;
+        cell.set(elf_resolver.into()).map_err(|_| {
+            Error::with_invalid_input(format!(
```

Comment:
"Invalid input" seems confusing; there is nothing wrong with the input. Can we use https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.AlreadyExists instead? Would be good to also check for that in the test.